nux/store/selectors.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import { includes, difference, keys } from 'lodash';
+
+/**
+ * Returns an object describing the guide, if any, that the given tip is a part
+ * of.
+ *
+ * @param {Object} state Global application state.
+ * @param {string} tipID The tip to query.
+ *
+ * @typedef {Object} NUX.GuideInfo
+ * @property {string[]} tipIDs       Which tips the guide contains.
+ * @property {?string}  currentTipID The guide's currently showing tip.
+ * @property {?string}  nextTipID    The guide's next tip to show.
+ *
+ * @return {?NUX.GuideInfo} Information about the associated guide.
+ */
+export function getAssociatedGuide( state, tipID ) {
+	for ( const tipIDs of state.guides ) {
+		if ( includes( tipIDs, tipID ) ) {
+			const nonDismissedTips = difference( tipIDs, keys( state.preferences.dismissedTips ) );
+			const [ currentTipID = null, nextTipID = null ] = nonDismissedTips;
+			return { tipIDs, currentTipID, nextTipID };
```

Comment:
Note: This function returns a new object every time it's called, meaning it will force React to reconcile every render if used directly in a component. This is not the case currently, but noting as a potential future performance concern (`createSelector` is used elsewhere as a memoization technique). Or we don't return an object and instead create separate selectors for each property returned here.